aiida/engine/processes/functions.py

```diff
@@ -140,18 +141,30 @@ def run_get_node(*args, **kwargs):
 
             process = process_class(inputs=inputs, runner=runner)
 
-            def kill_process(_num, _frame):
-                """Send the kill signal to the process in the current scope."""
-                LOGGER.critical('runner received interrupt, killing process %s', process.pid)
-                process.kill(msg='Process was killed because the runner received an interrupt')
+            # Only add handlers for interrupt signal to kill the current process if we are not in a daemon runner
```

Comment:
maybe add what the alternative is (i.e. if you are not in a daemon runner, what type of runner are you in then?).